recipes/cminpack/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools import files
+from conan import ConanFile
+import os
+
+required_conan_version = ">=1.45.0"
+
+
+class CMinpackConan(ConanFile):
+    name = "cminpack"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "About A C/C++ rewrite of the MINPACK software (originally in FORTRAN)" \
+                  "for solving nonlinear equations and nonlinear least squares problems"
+    topics = ("nonlinear", "solver")
+    homepage = "http://devernay.free.fr/hacks/cminpack/"
+    license = "BSD-like"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_EXAMPLES"] = "OFF"
+        tc.variables["CMINPACK_LIB_INSTALL_DIR"] = "lib"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.generate()
+
+    def layout(self):
+        cmake_layout(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+        # cminpack is a c library
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        files.copy(self, "CopyrightMINPACK.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        files.rmdir(self, os.path.join(self.package_folder, "share")) # contains cmake config files
+
+    def _library_postfix(self):
+        postfix = ""
+        if not self.options.shared:
+            postfix += "_s" # for static
+        if self.settings.build_type == "Debug":
+            postfix += "_d"
+
+        return postfix
+
+    def package_info(self):
+        minpack_include_dir = os.path.join("include", "cminpack-1")
+        
+        # the double precision version
+        self.cpp_info.libs = ['cminpack' + self._library_postfix()]
+        self.cpp_info.includedirs.append(minpack_include_dir)
```

Comment:
```suggestion
        self.cpp_info.components["_cminpack"].libs = ['cminpack' + self._library_postfix()]
        self.cpp_info.components["_cminpack"].includedirs.append(minpack_include_dir)
        self.cpp_info.components["_cminpack"].set_property("cmake_target_name", "cminpack::cminpack")

        self.cpp_info.components["_cminpack"].names["cmake_find_package"] = "cminpack"
        self.cpp_info.components["_cminpack"].names["cmake_find_package_multi"] = "cminpack"

```
Here, you create a new component named `_cminpack`, but you say to use specific names for cmake targets, which is the same from the project name, and it only uses the double precise library

Probably you will need to update system_libs, frameworks and cpp_info.defines for each component, as Conan does not like to mix general cpp_info with components.

Reply:
I committed your proposed changes. It fixes linking the correct library in case of the test_v1_package but still links the wrong library in the double precision test case. 

It seems, that the single precision component is still associated to the default cminpack::cminpack target. Is there any way to clear the default target or remove a component from the default target?

Reply:
This is the output from the test executables from the v1 package:

```
cminpack/1.3.8@test/testing (test package): Running test()

      fvec

           -1.181606      -1.429655      -1.606344
           -1.745269      -1.840654      -1.921586
           -1.984141      -2.022537      -2.468977
           -2.827562      -3.473582      -4.437612
           -6.047662      -9.267761      -18.91806
      fvecp - fvec

       -7.724666e-09  -3.432406e-09  -2.034843e-10
        2.313685e-09   4.331078e-09   5.984096e-09
        7.363281e-09    8.53147e-09   1.488591e-08
         2.33585e-08   3.522012e-08   5.301255e-08
         8.26666e-08   1.419747e-07    3.19899e-07
      err

           0.1141397     0.09943516     0.09674474
          0.09980447      0.1073116      0.1220445
           0.1526814              1              1
                   1              1              1
                   1              1              1

      fvec

           -1.181606      -1.429655      -1.606344
           -1.745269      -1.840654      -1.921586
           -1.984141      -2.022537      -2.468976
           -2.827562      -3.473582      -4.437612
           -6.047661      -9.267761      -18.91806
      fvecp - fvec

       -0.0001790524  -7.963181e-05  -4.768372e-06
        5.352497e-05   9.989738e-05   0.0001385212
        0.0001702309   0.0001974106   0.0003447533
        0.0005407333   0.0008158684    0.001227856
         0.001914024    0.003288269    0.007408142
      err

           0.2581466      0.2248907              0
            0.225696      0.2429258      0.2760358
           0.3457333              1              1
                   1              1              1
                   1              1              1
``` 

Note the difference in the second column, where the residual is 1e-9 for the double precision case, and ~1e-5 for the single precision case.

If I run the test_package (v2), I get exactly the same output (i.e. the single precision library is linked in both cases):

```
cminpack/1.3.8@test/testing (test package): Running test()
Capturing current environment in C:\src\conan-center-index\recipes\cminpack\all\test_package\build\generators\deactivate_conanrunenv-release-x86_64.bat
Configuring environment variables

      fvec

           -1.181606      -1.429655      -1.606344
           -1.745269      -1.840654      -1.921586
           -1.984141      -2.022537      -2.468976
           -2.827562      -3.473582      -4.437612
           -6.047661      -9.267761      -18.91806
      fvecp - fvec

       -0.0001790524  -7.963181e-05  -4.768372e-06
        5.352497e-05   9.989738e-05   0.0001385212
        0.0001702309   0.0001974106   0.0003447533
        0.0005407333   0.0008158684    0.001227856
         0.001914024    0.003288269    0.007408142
      err

           0.2581466      0.2248907              0
            0.225696      0.2429258      0.2760358
           0.3457333              1              1
                   1              1              1
                   1              1              1
Capturing current environment in C:\src\conan-center-index\recipes\cminpack\all\test_package\build\generators\deactivate_conanrunenv-release-x86_64.bat
Configuring environment variables

      fvec

           -1.181606      -1.429655      -1.606344
           -1.745269      -1.840654      -1.921586
           -1.984141      -2.022537      -2.468976
           -2.827562      -3.473582      -4.437612
           -6.047661      -9.267761      -18.91806
      fvecp - fvec

       -0.0001790524  -7.963181e-05  -4.768372e-06
        5.352497e-05   9.989738e-05   0.0001385212
        0.0001702309   0.0001974106   0.0003447533
        0.0005407333   0.0008158684    0.001227856
         0.001914024    0.003288269    0.007408142
      err

           0.2581466      0.2248907              0
            0.225696      0.2429258      0.2760358
           0.3457333              1              1
                   1              1              1
                   1              1              1
```